tests/ui/test_pv_encryption_ui.py

```diff
@@ -0,0 +1,291 @@
+            backend_path=self.vault_resource_name,
+            namespace=self.vault_resource_name,
+        )
+        logger.info("Storage Class Created via UI")
+
+        # Calling the Teardown Factory Method to make sure Storage Class is deleted
+        sc_obj = ocs.OCS(
+            kind=constants.STORAGECLASS, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        sc_obj._name = sc_name
+        teardown_factory(sc_obj)
+
+        # Verifying storage class details via UI
+        logger.info("Verifying Storage Class Details via UI")
+        verify_storage_class_ui(setup_ui, sc_name=sc_name)
+        logger.info("Storage Class Details Verified")
+
+        # Create ceph-csi-kms-token in the tenant namespace
+        logger.info("Creating ceph-csi-kms-token")
+        self.vault.vault_path_token = self.vault.generate_vault_token()
+        self.vault.create_vault_csi_kms_token(namespace=self.pro_obj.namespace)
+        logger.info("ceph-csi-kms-token created")
+
+        pvc_ui_obj = PvcUI(setup_ui)
+
+        access_mode = ["ReadWriteMany", "ReadWriteOnce"]
+        pvc_size = "5"
+        vol_mode = "Block"
+        for mode in access_mode:
+            pvc_name = create_unique_resource_name(
+                resource_description="test", resource_type="pvc"
+            )
+            pvc_ui_obj.create_pvc_ui(
+                project_name, sc_name, pvc_name, mode, pvc_size, vol_mode
+            )
+
+            logger.info("PVC Created via UI")
+
+            pvc_objs = get_all_pvc_objs(namespace=project_name)
+            pvc = [pvc_obj for pvc_obj in pvc_objs if pvc_obj.name == pvc_name]
+
+            assert pvc[0].size == int(pvc_size), (
+                f"size error| expected size:{pvc_size}"
+                f"\n actual size:{str(pvc[0].size)}"
+            )
+
+            assert pvc[0].get_pvc_access_mode == mode, (
+                f"access mode error| expected access mode:{mode} "
+                f"\n actual access mode:{pvc[0].get_pvc_access_mode}"
+            )
+
+            assert pvc[0].backed_sc == sc_name, (
+                f"storage class error| expected storage class:{sc_name} "
+                f"\n actual storage class:{pvc[0].backed_sc}"
+            )
+
+            assert pvc[0].get_pvc_vol_mode == vol_mode, (
+                f"volume mode error| expected volume mode:{vol_mode} "
+                f"\n actual volume mode:{pvc[0].get_pvc_vol_mode}"
+            )
+
+            # Verifying PVC via UI
+            logger.info("Verifying PVC Details via UI")
+            pvc_ui_obj.verify_pvc_ui(
+                pvc_size=pvc_size,
+                access_mode=mode,
+                vol_mode=vol_mode,
+                sc_name=sc_name,
+                pvc_name=pvc_name,
+                project_name=project_name,
+            )
+            logger.info("PVC Details Verified via UI")
+
+        # Verify if the key is created in Vault
+        logger.info("Verifying if the key is created in Vault")
```

Comment:
In OCS 4.7, the key in vault gets created after the PVC is attached to a pod, whereas in 4.8 the key gets created once the PVC is provisioned. It might be better to move the key validation check after the pod creation to accommodate both the scenarios.

Reply:
moved